### ISSUE TITLE
Add progress of installations (Snaps and Debs)

### ISSUE THREAD

**Issue by alculete** [ubuntu-flutter-community/software#906]
I think having a way to know how much longer my apps will take to install would be a good idea. Some connections in some countries are not very fast, so knowing the progress in percentage of how the download is going would be great.
Also for snaps, include the information on plugs being downloaded so the user knows when the app has completed to install everything.

**Comment by Feichtmeier** [comment]
@d-loose 

I often played around with this but never found a good solution. Snapd changes are divided into tasks. And you need to find out the time by

```dart
/// The time this change started.
final DateTime spawnTime;

/// The time this change completed.
final DateTime? readyTime;
```

Needs changes in snap model.

I think for package kit is this a bit easier.

@d-loose could you tackle this some day?

**Comment by d-loose** [comment]
Why would we need those times? Both the snapd and packagekit service report a progress status for their actions(`PackageKitItemProgressEvent`, `SnapdTaskProgress`), which we currently don't display on the `AppPage`. Adding those together with some more useful information about what packagekit is currently doing (besides `processing` and being `ready`) would definitely be very useful. I can take care of that, but will probably need some design feedback.

**Comment by Feichtmeier** [comment]
Snapd has the kind of change in there and I think I put this into the snap model. Again those are strings you need to check like currently in the snap model 

```dart
 String getChangeMessage({
    required BuildContext context,
    required String? changeKind,
  }) {
    switch (changeKind) {
      case 'install-snap':
        return context.l10n.installing;
      case 'remove-snap':
        return context.l10n.removing;
      case 'refresh-snap':
        return context.l10n.refreshing;
      case 'connect-snap':
        return context.l10n.changingPermissions;
      case 'disconnect-snap':
        return context.l10n.changingPermissions;
      default:
        return '';
    }
  }
  ```

which could be made more elegant like putting it as an extension on snapdchange or even directly into snapd.dart

**Comment by d-loose** [comment]
Fixed by #916 #944
